core/services/workflows/syncer/workflow_registry.go

```diff
@@ -256,8 +280,31 @@ func (w *workflowRegistry) Name() string {
 	return name
 }
 
-// readRegistryEvents polls the contract for events and send them to the events channel.
-func (w *workflowRegistry) readRegistryEvents(ctx context.Context, eventTypes []WorkflowRegistryEventType, don capabilities.DON, reader ContractReader, lastReadBlockNumber string) {
+func (w *workflowRegistry) handleEventLoop(ctx context.Context) {
+	defer w.wg.Done()
+	w.lggr.Debug("running handleEventLoop")
+	for {
+		select {
+		case <-ctx.Done():
+			w.lggr.Debug("shutting down handleEventLoop")
+			return
+		case event, open := <-w.eventCh:
```

Comment:
Will this ever happen? I don't see us closing eventCh anywhere

I think it's fine to simplify here and just rely on ctx.Done() which is derived from the service stopCh